### PR TITLE
DDBCMS-358 - Subsearch: Improvements to modules.

### PR DESCRIPTION
**Pull request by bozeak**
https://inlead.atlassian.net/browse/DDBCMS-358

### CODE REVIEW

File: ting_subsearch_secondary.module [hunk 1]
```diff
@@ -99,46 +99,51 @@ function ting_subsearch_secondary_execute_secondary_search($keys, $results) {
...
+    if ($ratio > variable_get('ting_subsearch_secondary_ps_factor', 2)) {
+
+      $items = [];
+      $number_of_records_to_show = variable_get('ting_subsearch_secondary_number_of_results', 4);
+      foreach ($secondary_result->collections as &$collection) {
+        if ($number_of_records_to_show == 0) {
+          continue;
+        }
+        else {
+          $number_of_records_to_show--;
```
Comment (guddo) [inline ting_subsearch_secondary.module]: These two dashes on purpose?

Comment (OIOIOOOI) [reply ting_subsearch_secondary.module]: Two dashes decrements the value by one.

File: ting_subsearch_secondary.module [hunk 2]
```diff
@@ -99,46 +99,51 @@ function ting_subsearch_secondary_execute_secondary_search($keys, $results) {
...
+    $ratio = $secondary_result->numTotalObjects / $results->getNumTotalObjects();
+    if ($ratio > variable_get('ting_subsearch_secondary_ps_factor', 2)) {
+
+      $items = [];
+      $number_of_records_to_show = variable_get('ting_subsearch_secondary_number_of_results', 4);
+      foreach ($secondary_result->collections as &$collection) {
+        if ($number_of_records_to_show == 0) {
+          continue;
+        }
+        else {
```
Comment (OIOIOOOI) [inline ting_subsearch_secondary.module]: There's no need for an `else` here when a `continue` is used in the conditional.

File: ting_subsearch_secondary.module [hunk 3]
```diff
@@ -99,46 +99,51 @@ function ting_subsearch_secondary_execute_secondary_search($keys, $results) {
   ]);
 
   if ($secondary_result->numTotalObjects > variable_get('ting_subsearch_secondary_result_min', 100)) {
-    $items = [];
+    $ratio = $secondary_result->numTotalObjects / $results->getNumTotalObjects();
+    if ($ratio > variable_get('ting_subsearch_secondary_ps_factor', 2)) {
+
```
Comment (OIOIOOOI) [inline ting_subsearch_secondary.module]: Empty line.

File: ting_subsearch_secondary.module [hunk 4]
```diff
@@ -99,46 +99,51 @@ function ting_subsearch_secondary_execute_secondary_search($keys, $results) {
...
+      foreach ($secondary_result->collections as &$collection) {
+        if ($number_of_records_to_show == 0) {
+          continue;
+        }
+        else {
+          $number_of_records_to_show--;
+        }
+        $build = ting_collection_view($collection, 'minimal');
+        $markup = drupal_render($build);
+        $return = '<a target="_blank" href="https://bibliotek.dk/work/' . $collection->ding_entity_id . '">' . $markup . '</a>';
```
Comment (OIOIOOOI) [inline ting_subsearch_secondary.module]: Any reason not to use `l()` or a render array here?

**Review by OIOIOOOI — CHANGES_REQUESTED** [review]

File: ting_subsearch_secondary.module (hunk 1 above)
Comment (OIOIOOOI) [reply ting_subsearch_secondary.module]: Not related to PR - I see that original implementation uses this variable as a counter to how many times to render an object, yet I wonder why it wasn't implemented with a `brake` statement instead when `$number_of_records_to_show` reaches zero or use a more generic `for` loop.

**Review by OIOIOOOI — APPROVED** [review]

**Pull request merged**